AppCenterCrashes/AppCenterCrashes/MSCrashes.mm

```diff
@@ -317,6 +317,14 @@ - (void)applyEnabledState:(BOOL)isEnabled {
       [crashSetupDelegate didSetUpCrashHandlers];
     }
 
+    // Set up lifecycle event handler.
+#if !TARGET_OS_OSX
+    [MS_NOTIFICATION_CENTER addObserver:self
```

Comment:
This is one-time thing, can be done at the time Crashes is started.

Reply:
Remove the observer when it processed crashes. Doesn't need to keep it.

Reply:
I would like to keep centralized removing observers logic. Also, that way anyway requires re-adding it after re-enable since we clean all observers on disable.
I don't thing that it has any perf impart.

Reply:
> I would like to keep centralized removing observers logic.

I don't see it is removed on disable.

> Also, that way anyway requires re-adding it after re-enable since we clean all observers on disable.

When it is enabled (from disabled state), we don't need to add it again. We know there is no crashes stored.

> I don't thing that it has any perf impart.

Even though it has no perf impact, we don't need to listen to it in that case.

Reply:
https://github.com/microsoft/appcenter-sdk-apple/pull/1864/files#diff-60b2ff58c9ed4c6302886496bee39aa5R294